src/qt/nametablemodel.cpp

```diff
@@ -0,0 +1,537 @@
+    for (std::vector<CTxOut>::const_iterator it(vout.begin()); it != vout.end(); ++it)
+    {
+        if (!CNameScript::isNameScript(it->scriptPubKey))
+        {
+            continue;
+        }
+
+        CNameScript nameScript(it->scriptPubKey);
+        switch (nameScript.getNameOp ())
+        {
+            case OP_NAME_NEW:
+                break;
+
+            case OP_NAME_FIRSTUPDATE:
+            case OP_NAME_UPDATE:
+                priv->refreshName(nameScript.getOpName ());
+                break;
+
+            default:
+                assert (false);
+        }
+    }
+
+}
+
+int NameTableModel::rowCount(const QModelIndex &parent /*= QModelIndex()*/) const
+{
+    Q_UNUSED(parent);
+    return priv->size();
+}
+
+int NameTableModel::columnCount(const QModelIndex &parent /*= QModelIndex()*/) const
+{
+    Q_UNUSED(parent);
+    return columns.length();
+}
+
+QVariant
+NameTableModel::data(const QModelIndex &index, int role) const
+{
+    if (!index.isValid())
+        return QVariant();
+
+    NameTableEntry *rec = static_cast<NameTableEntry*>(index.internalPointer());
+
+    if (role == Qt::DisplayRole || role == Qt::EditRole)
+    {
+        switch(index.column())
+        {
+        case Name:
+            return rec->name;
+        case Value:
+            return rec->value;
+        case ExpiresIn:
+            if (!rec->HeightValid()) {
+                return QVariant();
+            }
+            else {
+                int nBestHeight = chainActive.Height();
+                // OG: return rec->nHeight + GetDisplayExpirationDepth(rec->nHeight)
+                // - pindexBest->nHeight;
+                return rec->nHeight + 36000 - nBestHeight;
+            }
+        }
+    }
+    return QVariant();
+}
+
+QVariant
+NameTableModel::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if (orientation == Qt::Horizontal)
+    {
+        if (role == Qt::DisplayRole)
+        {
+            return columns[section];
+        }
+        else if (role == Qt::TextAlignmentRole)
+        {
+            return column_alignments[section];
+        }
+        else if (role == Qt::ToolTipRole)
+        {
+            switch(section)
+            {
+            case Name:
+                return tr("Name registered using Namecoin.");
+            case Value:
+                return tr("Data associated with the name.");
+            case ExpiresIn:
+                return tr("Number of blocks, after which the name will expire."
+                          " Update name to renew it.\nEmpty cell means pending"
+                          " (awaiting automatic name_firstupdate or awaiting "
+                          "network confirmation).");
+            }
+        }
+    }
+    return QVariant();
+}
+
+Qt::ItemFlags
+NameTableModel::flags(const QModelIndex &index) const
+{
+    if (!index.isValid())
+        return 0;
+    //NameTableEntry *rec = static_cast<NameTableEntry*>(index.internalPointer());
```

Comment:
Please remove the commented-out code, or clarify why it is still relevant (e. g., with a `FIXME`).